craftcms/cms
# Bugfix/16158 creating new nested entry in non primary site

### Description
How to reproduce:
- have at least 2 sites set up
- create a channel section with the propagation method set to “all” enabled for both sites;
- with an entry type that contains a matrix field
- the matrix field is in the cards view mode with the propagation method set to “none”
- go to the non-primary site, create an entry in that channel and try to add a new nested element; 
- notice “Attempting to save an element in an unsupported site.” error

Reverted the change from #16136 to fix the reported issue.

Added a different fix for #16136.

### Related issues
#16158 
#16144 
#16176
#16196
